integrations/transportation/ryanair.js

```diff
@@ -0,0 +1,135 @@
+import moment from 'moment';
+import request from 'superagent';
+import { ACTIVITY_TYPE_TRANSPORTATION, TRANSPORTATION_MODE_PLANE } from '../../definitions';
+
+const agent = request.agent();
+const BASE_URL = 'https://api.ryanair.com/userprofile/rest/api/v1/';
+const LOGIN_URL = `${BASE_URL}login`;
+const PROFILE_URL = `${BASE_URL}secure/users/`;
+// user info available at `${PROFILE_URL}${customerId}/profile/full/`
+
+async function logIn(username, password) {
+  const res = await agent
+    .post(LOGIN_URL)
+    .type('application/x-www-form-urlencoded')
+    .set('Authorization', 'Basic ')
+    .send({
+      password,
+      username,
+    });
+
+  if (!res.ok) {
+    console.error('--------RESPONSE---------', res);
+    throw Error('Error while logging in.');
+  }
+
+  return {
+    token: res.body.token,
+    customerId: res.body.customerId,
+  };
+}
+
+async function connect(requestLogin, requestWebView) {
+  // Here we can request credentials etc..
+
+  // Here we can use two functions to invoke screens
+  // requestLogin() or requestWebView()
+  const { username, password } = await requestLogin();
+
+  if (!(password || '').length) {
+    throw Error('Password cannot be empty');
+  }
+
+  return logIn(username, password);
+}
+
+function disconnect() {
+  // Here we should do any cleanup (deleting tokens etc..)
+  return {};
+}
+
+async function getAllFlights(entries, customerId, token) {
+  const allFlights = [];
+
+  await Promise.all(entries.map(async (entry) => {
+    await agent
+      .put(`${PROFILE_URL}${customerId}/bookings/booking`)
+      .set('Accept', 'application/json')
+      .set('X-Auth-Token', token)
+      .send({
+        bookingId: entry.bookingId,
+        pnr: entry.pnr,
+      })
+      .then((res) => {
+        if (res.ok) {
+          res.body.Flights.forEach((singleFlight) => {
+            allFlights.push({ 
+              bookingId: res.body.BookingId,
+              flightInfo: singleFlight,
+              pnr: res.body.RecordLocator,
+            });
+          });
+        }
+      });
+  }));
+  return allFlights;
+}
+
+async function collect(state) {
+  const { token, customerId } = state;
+
+  const pastBookings = await agent
+    .get(`${PROFILE_URL}${customerId}/bookings/past`)
+    .set('Accept', 'application/json')
+    .set('X-Auth-Token', token);
+
+  if (!pastBookings.ok) {
+    console.error('--------RESPONSE---------', pastBookings);
```

Comment:
use `logger.logDebug()` in order to use the in-app logger.
I think here you can just remove this line as it seems only to be relevant when developing the integration

Reply:
Oh, you're right, forgot to use logger. Btw, should it be used along with throwing an error? Just added logger, and left the error throw in for now